### DISCUSSION TITLE
Public Apps planed?

### DISCUSSION THREAD

**Discussion by MasterOfWeb** [Budibase/budibase#3866] [Ideas]
Hi
I'm trying to build a public SAAS-App and was looking at budibase as an option only to finde out, it's only for internal apps.

I really love how easy it is to build crud-screens, automations and so on. It would be perfect for my App. But why can't I uses this technology for other things than internal apps? The only main difference is the way user registration and roles are handled, right?

Do you have any plans on changing this in the future?

**Comment by mjashanks** [comment]
If you can do it with Budibase - go ahead! It's just that this won't be our focus, so certain features that you might need won't get done if they don't apply to Business apps.

There's also a bit of a distinction to be made on "Internal", "Public" and "SaaS".  There's no reason why you can't build a "Public" app with Budibase - for example, you might want to give access to people outside of your organisation. However, the app is ultimately for your organisation's use. 

The model that we aren't supporting is where you create your own software product, that you are selling access to - i.e. a SaaS Product.

- Self-registration
- Subscriptions
- Multi-tenancy for a single app

These are all things that you may need to build a SaaS product - there may be more.

**Reply by xieciun** [reply]
Hi there, 

Can you please elaborate a bit more on why bb shouldn't (can't) be used for such scenarios? Is it a matter of capabilities or the bb licence? I was wondering if I can use it to build a kind of landing page for my SaaS (the main part will be created elsewhere).

**Reply by universenz** [reply]
Yes, please elaborate on (why) and (if) there are ever plans to expand support for self registration, subscriptions, and/or multi-tenancy. Is it a technical limitation for the team? Is it a financial limitation? Some insight would be appreciated.

**Reply by mjashanks** [reply]
@xieciun  - there is no reason why you can't use Budibase for this. Please go ahead and try. It's just that this is not our focus - so you might find that we would not add the features that you need to build this sort of thing.

@universenz 
- Self-registration - maybe someday, but it's not on the roadmap right now.
- Subscriptions - can't see this ever happening.
- Multi-tenancy (for a single app) - can't see this happening. We do support multi-tenancy for an entire deployment of Budibase (i.e. lots of apps in a tenant), on our enterprise plan

Again, we just don't see demand for this among our target users - building business apps that are primarily for running the operations of a single company. These are mostly internal apps, although many use cases are external facing.

There is no technical or financial limitation - it's philosophical. It just doesn't fit with the direction of the product. 

I am very happy to hear some use cases for the above features that apply to our intended target users. If you want these features because you are trying to build your own SaaS product with Budibase - then the answer is no.

**Reply by xieciun** [reply]
@mjashanks Thank you :)

**Reply by yashness** [reply]
I'd also like to request such feature. It's very powerful, only thing missing is self-registration & a single app which could be hosted on the domain & accessed by registered users. In fact, if it could be done as a public app (view only) without registration, that would be super cool!

**Reply by kulkarni005** [reply]
You can do it by Wrapping Budibase Public APIs through your SAAS - App which itself is getting created / Managed by Budibase.
inception !!!!

**Reply by maverjk** [reply]
@kulkarni005 Can you elaborate how in a detailed way?

**Comment by kulkarni005** [comment]
You can do it by Wrapping Budibase Public APIs through your SAAS - App which itself is getting created / Managed by Budibase.
inception !!!!
